### DISCUSSION TITLE
The next ...

### DISCUSSION THREAD

**Discussion by ikedas** [sympa-community/sympa#1629] [Release works]
Branch for Release 6.2.70 sympa-community/sympa#1384 was created.

If anyone wants to release the next version anytime, please feel free to do it.

**Comment by racke** [comment]
What about beta releases? Or we just go with the bug fixes?

**Comment by ikedas** [comment]
Beta releases are desirable, if they are meaningful.

Needless to say, all changes described in the release notes should be tested well. --- If they had been, we would not have had to [completely revert the feature added in the two previous release](/sympa-community/sympa/issues/1359).

Anyway, I have stepped down as release manager (That's why I haven't set up milestone tag 6.2.70 or alike).  So @racke, please feel free to make suggestions for future releases.

**Comment by racke** [comment]
I guess it is up to me to pick up the baton as release manager. My idea is that we shouldn't wait too long for the 6.2.70 in order to fix the bugs introduced with 6.2.68 which bypassed our Potemkin beta tester team.

@ikedas Please send me instructions about how to release Sympa betas / new versions.

**Comment by ikedas** [comment]
Hi @racke,

Potemkin? ...Ah well, well, I'm convinced they will rise up in the end, but now seems not the time yet (however you probably are not talking about the battleship:smile:).

----
The instruction is [here](https://github.com/sympa-community/sympa/blob/sympa-6.2/support/README.support.md) in "How to prepare a new source tarball".  Some additional information:
  * To commit changes, `Sympa authors <sympa-developpers@listes.renater.fr>` (`p` is duplicated) is used as the author at all.
  * Item 3 (Retrieve latest translations and update translation) may not be necessary this time.
  * Item 10 (Add tag to remote repository)  will be automatically done by publishing a new [release](https://github.com/sympa-community/sympa/releases) page.  Conversely, all changes must have been pushed to the remote repository at the time the release page will be published.
  * Afterward, you should post release announcements to `sympa-announce@listes.renater.fr` _and_ `lets@translate.sympa.org` (both of them are moderated).
  * After the all steps above, there is the task of pushing the release page to the [website](https://github.com/sympa-community/sympa-community.github.io/tree/master/_posts), which I will do, as it has not yet been automated.

Please let me know if I've left anything out.

**Comment by racke** [comment]
@ikedas reference was to Potemkin's villages where all houses are just the front without any living quarters to show up to visitors :grimacing:

**Comment by ikedas** [comment]
I just learned it from Wikipedia yesterday.  Well, person in power is often ignorant of the truth. :upside_down_face:

**Comment by ikedas** [comment]
I'd like to solve at least these two before the next release (we already know how to fix both of them).

  * sympa-community/sympa#1410
  * sympa-community/sympa#1407

----
These may be suspended to the next-next release.

  * sympa-community/sympa#1386 (PRs have been submitted)
  * Alternative PR for sympa-community/sympa#903

I hope to resolve these asap but have not started yet.

  * sympa-community/sympa#1354 (I've not yet found the solution)
  * sympa-community/sympa#1242 (we have to find the way to handle errorneous topics.conf)
  * sympa-community/sympa#1124 (mailto link should be replaced with link to web GUI if possible)

**Comment by racke** [comment]
I created milestones for 6.2.70 and 6.2.72 and assigned them to these issues.

**Comment by ikedas** [comment]
Thanks @racke !

But the last three in above may not be solved soon, so I moved them to 6.2.72....
And I added issues already closed to 6.2.70 for memory.

**Comment by racke** [comment]
So my tentative plan would be for the release:

- Test run / Automate release process
- Write email to list about change of release management, pending release and available beta testers
- Set release date base on feedback from the list
- Coordinate release with package builds

**Comment by ikedas** [comment]
Nice plan!

> Test run / Automate release process

Currently test cases run by each push with [a workflow](https://github.com/sympa-community/sympa/blob/sympa-6.2/.github/workflows/make-check.yml) of [GitHub Actions](https://github.com/features/actions) (lacks many cases, though:unamused:).
I don't think we necessarily have to use GitHub Actions, but actions required to [publish release](https://github.com/marketplace/actions/publish-release) seem already available.

**Comment by ikedas** [comment]
@everyone (particularly @racke ),

I personally think that none of the issues currently open need to be resolved before the next stable release.

  * Should we declare a code freeze?
    I was pretty vague on this procedure, but it should have been done if possible.
  * Will the next release be a beta or a directly stable release?

I will post about automating the release process in another thread.

**Comment by racke** [comment]
Code freeze would mean to me that we don't update sympa-6.2 with the exception of critical problems and translations.

As we did mostly bug fixes, I would prefer to make a stable release without beta.

**Comment by ikedas** [comment]
@racke (perhaps you author the release announcement),

I wrote a notice about the transition of web site and the lists. Please include this text in the next release announcement.

I am not a proper English writer, so please fix anything odd.

---

## The community will have a new domain

The web site for the community of Sympa, which has been operating under the sympa.org domain, will now move to the new domain sympa.community.

After [date] in [month], the official website URL will be:

> https://www.sympa.community

(Note: sympa-community.github.io, previously used URL, will be automatically redirected to the new site.)

In addition, the mailing lists for the community, which have been hosted by RENATER on their servers, will be migrated to the new domain. We will announce again when the transition is ready.

We would like to thank RENATER for hosting the website and mailing lists for two decades.

  * Contact

    For inquiries regarding transition of web and lists, please contact devel@sympa.community.


The Sympa Community

---

**Comment by ikedas** [comment]
Hi @racke,
Will the fix to sympa-community/sympa#1454 be ready for 6.2.70? I think we can merge it in the next few days, so please add it if possible.

**Comment by racke** [comment]
> Hi @racke,
> Will the fix to sympa-community/sympa#1454 be ready for 6.2.70? I think we can merge it in the next few days, so please add it if possible.

Thanks for the quick fix. Added the PR to the 6.2.70 milestone.

**Comment by ikedas** [comment]
@racke, when is the release of 6.2.70 likely to be?

I will have some free time on the week of 15th Aug., so I'l try to migrate the remainder of lists then. If the release is later than that, I'll make announcements by my own.

**Comment by ikedas** [comment]
@racke, since you seem to be busy (or maybe on vacation :-)), I will make an announcement about mailing list migration today or tomorrow by myself (migration will start after the afternoon UTC of 15th Aug). You can proceed with the release separately.

 If you are having trouble automating release process, --- it's hard to automate everything at once, so we may just work manually at first and then gradually automate it.)

**Comment by ikedas** [comment]
@racke,

This is a fix to bug injected by previous PR.  I think this should be included in 6.2.70 (I hope this is the last one...).

  * sympa-community/sympa#1462

Addition: Use new domain name and list addresses.
  * sympa-community/sympa#1463
----
For now, following bugs / enhancements are ready for merging, however IMO they are not critical and may be labelled 6.2.72.

  * sympa-community/sympa#1429 (confusing err log in inclusion)

  * sympa-community/sympa#1434 (loading CSV driver crashes)

  * sympa-community/sympa#1441 (add permalink on messages in web archive)

  * sympa-community/sympa#1447 (reduce header fields shown on archives)

  * A group of PRs enhancing ARC feature:
      * sympa-community/sympa#1449
      * sympa-community/sympa#1450
      * sympa-community/sympa#1453
      * sympa-community/sympa#1457

  * sympa-community/sympa#1456 (fix broken SOAP schema)

**Comment by racke** [comment]
Moving to a new flat this month, so I would like to propose the weekend 2.-3. September for the 6.2.70 release.

**Comment by ikedas** [comment]
I agreee to include https://github.com/sympa-community/sympa/pull/1463 (Changes with new domain).
I would appreciate it if you could merge [translation branch](https://github.com/sympa-community/sympa/compare/translation) (should I submit a PR?).

**Comment by racke** [comment]
Please submit a PR for the translations.

**Comment by ikedas** [comment]
Done!

**Comment by racke** [comment]
Using "make tidyall" in my environment creates a huge commit: https://github.com/sympa-community/sympa/compare/sympa-6.2...racke:tidy/6.2.70?expand=1

@ikedas Can you elaborate on your Tidy setup?

**Comment by ikedas** [comment]
Perhaps the versions of  Test::PerlTidy and/or Perl::Tidy don’t match (see cpanfile),
Anuways this time , I think, skipping tidyall does not so matter.

**Comment by racke** [comment]
The versions are quite old:

     requires 'Test::PerlTidy', '== 20130104';
     requires 'Perl::Tidy', '== 20180220';

Skipping tidyall for this release.

**Comment by ikedas** [comment]
btw I was working on the release with a recent version of Perl compiled by myself instead of the one bundled to the OS; I added the --with-perl option to configure so that my own Perl would be used.

**Comment by ikedas** [comment]
I'll be away most of 17th to 19th.  If you will be working for release in this weekend, please feel free to do it without asking me.

---
BTW for now following issues/prs are ready for merging.  If no objection, It should be labelled _6.2.72_ (_not_ 6.2.70).

  * sympa-community/sympa#1429 (confusing err log in inclusion)

  * sympa-community/sympa#1434 (loading CSV driver crashes)

  * sympa-community/sympa#1441 (add permalink on messages in web archive)

  * sympa-community/sympa#1447 (reduce header fields shown on archives)

  * A group of PRs enhancing ARC feature:
      * sympa-community/sympa#1449
      * sympa-community/sympa#1450
      * sympa-community/sympa#1453
      * sympa-community/sympa#1457

  * sympa-community/sympa#1456 (fix broken SOAP schema)

  * sympa-community/sympa#1466 (bug in web UI)

  * sympa-community/sympa#1467 (update SMTP status codes used in bouncers page)
 
  * sympa-community/sympa#1470 (spam_status feature: clarify parameters and add scenarios)
 
  * sympa-community/sympa#1471 (remove unuseful OChangeLog file)

  * sympa-community/sympa#1474 (temlate error in creation of automatic list not detected)

  * sympa-community/sympa#1475 (clarify `(welcome|remind)_return_path` parameters)

**Comment by racke** [comment]
@ikedas Sorry, I don't understand your comment "It should be labelled 6.2.72 (not 6.2.70)." Can you elaborate please?

**Comment by ikedas** [comment]
I think these fixes/improvbements should be included in milestone 6.2.72, but not in 6.2.70, the next release.
So if no objection, I'd like to label these as 6.2.72.

**Comment by ikedas** [comment]
Since there are no objection, I did.
The reason is simple: if we don't label them, we'll miss them.

**Comment by racke** [comment]
Fine for me :+1:

**Comment by ikedas** [comment]
@racke, when would the new release be possible? If you don't have the time to do it, shall I switch work?

It has been 9 months since the last release.

**Comment by ikedas** [comment]
Can I prepare [draft of the release note](https://github.com/sympa-community/sympa/releases)?

**Comment by ikedas** [comment]
@racke, when would the new release be possible? If you don't have the time to do it, shall I switch work?

**Comment by racke** [comment]
Will do the release next Sunday.

On 2 October 2022 12:47:34 EEST, IKEDA Soji ***@***.***> wrote:
***@***.***, when would the new release be possible? If you don't have the time to do it, shall I switch work?
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/orgs/sympa-community/teams/everyone/discussions/30/comments/39
>You are receiving this because you were mentioned.
>
>Message ID: ***@***.***>
-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by ikedas** [comment]
Now all issues tagged 6.2.72 have had [on going] lael, i.e. have PR to solve their problems.

OTOH these are submitted and have [on going] label but have not had 6.2.72 tag:
  * sympa-community/sympa#1437 Correct parameters for CSV driver
  * sympa-community/sympa#1502 Update default of anonymous_header_fields parameter
  * sympa-community/sympa#1498 dmarc_protection.phrase "From" format INCORRECT
  * sympa-community/sympa#1497 WWSympa: Authentication mechanism that does not require login if the authentication module allows it
  * sympa-community/sympa#1494 accessing db_additional_subscriber_fields fields in footer vs. tt2 templates
  * sympa-community/sympa#1424 Deprecate conf_2_db feature
  * sympa-community/sympa#1488 [WWSympa] Cannot allow owners to manage editors of a distribution list
  * sympa-community/sympa#1483 Deprecate "dkim" authentication method for scenarios
  * sympa-community/sympa#1474 Template error parsing not detected

@racke, if you like, could you please tag  these?

**Comment by racke** [comment]
Done!
